lib/backend/syncersv1/felixsyncer/felixsyncer_e2e_test.go

```diff
@@ -155,7 +183,20 @@ var _ = testutils.E2eDatastoreDescribe("Felix syncer tests", testutils.Datastore
 					}
 				}
 				Expect(err).NotTo(HaveOccurred())
-
+				addCleanup(func() {
+					for i := 0; i < 5; i++ {
+						// This can fail due to an update conflict, so we allow a few retries.
+						node, err = c.Nodes().Get(ctx, "127.0.0.1", options.GetOptions{})
+						Expect(err).NotTo(HaveOccurred())
+						node.Spec.BGP = oldBGPSpec
+						node.Spec.VXLANTunnelMACAddr = oldVXLANTunnelMACAddr
+						node, err = c.Nodes().Update(ctx, node, options.SetOptions{})
+						if err == nil {
+							break
+						}
+					}
+					Expect(err).NotTo(HaveOccurred())
```

Comment:
Maybe I did not spot it last time. So What is the purpose of retry if there is `Expect(err).NotTo(HaveOccurred())`

Reply:
It checks that it succeeded after 1-5 attempts.

Reply:
My understanding is once expectation failed, ginkgo will start to complete the test case and run `after Each`, so it probably won't retry 5 times. Am I right?

Reply:
Oh, I see what you mean. Understood now.